src/migrations.rs

```diff
@@ -0,0 +1,288 @@
+        num_traits::FromPrimitive::from_i64(version)
+    }
+
+    /// Convert to the integer representation used as version in the database.
+    fn as_i64(self) -> i64 { self as i64 }
+
+    /// Whether introducing the database schema version is destructive, meaning
+    /// not backwards compatible.
+    /// Note: We use match statements here to catch missing variants at
+    /// compile-time. This enforces explicit evaluation when adding a new
+    /// database schema, ensuring awareness of whether the change is
+    /// destructive.
+    fn is_destructive(self) -> bool {
+        match self {
+            SchemaVersion::Empty => false,
+            SchemaVersion::InitialSchema => false,
+            SchemaVersion::PLTSchema => false,
+        }
+    }
+
+    /// Whether the database schema version is a partial migration.
+    /// Note: We use match statements here to catch missing variants at
+    /// compile-time. This enforces explicit evaluation when adding a new
+    /// database schema, ensuring awareness of whether the change is partial.
+    fn is_partial(self) -> bool {
+        match self {
+            SchemaVersion::Empty => false,
+            SchemaVersion::InitialSchema => false,
+            SchemaVersion::PLTSchema => false,
+        }
+    }
+
+    /// Run migrations for this schema version to the next.
+    async fn migration_to_next(
+        &self,
+        database_client: &mut DatabaseClient,
+    ) -> anyhow::Result<SchemaVersion> {
+        let start_time = chrono::Utc::now();
+        let new_version = match self {
+            SchemaVersion::Empty => {
+                database_client
+                    .as_ref()
+                    .batch_execute(include_str!("../resources/m0001-initial.sql"))
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "Failed to execute SQL schema version: {}",
+                            SchemaVersion::InitialSchema
+                        )
+                    })?;
+                SchemaVersion::InitialSchema
+            }
+            SchemaVersion::InitialSchema => {
+                database_client
+                    .as_ref()
+                    .batch_execute(include_str!("../resources/m0002-PLT.sql"))
+                    .await
+                    .with_context(|| {
+                        format!(
+                            "Failed to execute SQL schema version: {}",
+                            SchemaVersion::PLTSchema
+                        )
+                    })?;
+                SchemaVersion::PLTSchema
+            }
+            SchemaVersion::PLTSchema => unimplemented!(
+                "No migration implemented for database schema version {}",
+                self.as_i64()
+            ),
+        };
+        let end_time = chrono::Utc::now();
+        insert_migration(database_client, &new_version.into(), start_time, end_time).await?;
```

Comment:
I think we should run the above migration and this `insert_migration` in the same postgresql transaction, such that aborting/crashing during a migration results in a rollback.